packages/tanstack-react-query/src/internals/createOptionsProxy.ts

```diff
@@ -33,10 +33,48 @@ import {
   trpcSubscriptionOptions,
   type TRPCSubscriptionOptions,
 } from './subscriptionOptions';
-import type { QueryType, ResolverDef } from './types';
+import type { QueryType, ResolverDef, TRPCQueryKey } from './types';
 import { getQueryKeyInternal } from './utils';
 
+export interface DecorateQueryKeyable {
+  /**
+   * Calculate the Tanstack Query Key for a Route
+   *
+   * @see https://tanstack.com/query/latest/docs/framework/react/guides/query-keys
+   */
+  queryKey: () => TRPCQueryKey;
+
+  /**
+   * Calculate a Tanstack Query Filter for a Route
+   *
+   * @see https://tanstack.com/query/latest/docs/framework/react/guides/filters
+   */
+  queryFilter: () => QueryFilters;
```

Comment:
I'm still not convinced this is a very useful method, especially since it doesn't take any of the other query filter args (predicate functions etc). why not just use `{ queryKey: trpc.foo.queryKey() }`

Reply:
I think the idea would be that we have full control, so if we do want to set some of the other filters internally we can without breaking anybody's apps

I'm open to not having it, just felt like it's appropriate for a React Query factory to provide

Reply:
Could also add options to override some bits, but I didn't want to dive straight into that world, worth analysing some use cases

Reply:
I'd say until we have identified these use cases and added support for the other filter options just the queryKey suffices? Not a hill I'd die on so if you still see value in it go ahead and merge 👍